src/sentry/middleware/auth.py

```diff
@@ -1,7 +1,34 @@
 from __future__ import absolute_import
 
-from django.contrib.auth import middleware
+from django.contrib import auth
+from django.utils.functional import SimpleLazyObject
+from django.dispatch import receiver
+from sentry.utils.dates import to_datetime
+from sentry.utils.auth import bump_session_timestamp
 
 
-class AuthenticationMiddleware(middleware.AuthenticationMiddleware):
-    pass
+@receiver(auth.user_logged_in)
+def on_user_logged_in(sender, request, user, **extra):
+    if user.is_authenticated():
+        bump_session_timestamp(request)
+
+
+def get_user_safely(request):
+    user = auth.get_user(request)
```

Comment:
we should change our backend implementation of get_user to use get_from_cache since this lookup will happen pretty much all the time